ukat/mapping/mtr.py

```diff
@@ -0,0 +1,114 @@
+import os
+import numpy as np
+import nibabel as nib
+import warnings
+
+
+class MTR:
+    """
+    Generates a Magnetisation Transfer Ratio (MTR) map from a series of volumes
+    collected with 2 different magnetisation transfer values (ON and OFF).
+
+    Attributes
+    ----------
+    mtr_map : np.ndarray
+        The estimated magnetisation transfer ratio values.
+        Each value should be a float between 0 and 1.
+    shape : tuple
+        The shape of the MTR map.
+    mt_on : np.ndarray
+        The array corresponding to the magnetisation transfer value ON.
+    mt_off : np.ndarray
+        The array corresponding to the magnetisation transfer value OFF.
+    mask : np.ndarray
+        A boolean mask of the voxels to fit.
+    """
+
+    def __init__(self, pixel_array, affine, mask=None):
+        """Initialise a MTR class instance.
+
+        Parameters
+        ----------
+        pixel_array : np.ndarray
+            A 3D/4D array containing the image at each magnetisation transfer
+            value, i.e. the dimensions of the array are (x, y, 2) or
+            (x, y, z, 2). The assumption is that `mt_off` comes first in
+            `pixel_array` and `mt_on` comes second in `pixel_array`, i.e.
+            `pixel_array[..., 0]` corresponds to `mt_off` and
+            `pixel_array[..., 1]` corresponds to `mt_on`.
+        affine : np.ndarray
+            A matrix giving the relationship between voxel coordinates and
+            world coordinates.
+        mask : np.ndarray, optional
+            A boolean mask of the voxels to fit. Should be the shape of the
+            desired MTR map rather than the raw data i.e. omit the on/off
+            dimension.
+        """
+        assert (pixel_array.shape[-1] == 2), 'The input should contain 2 MT ' \
+                                             'values (ON / OFF). The last ' \
+                                             'dimension of the input ' \
+                                             'pixel_array must be 2.'
+        if np.sum(pixel_array[..., 1]) >= np.sum(pixel_array[..., 0]):
+            warnings.warn(f'The sum of all MT_ON values should be greater '
```

Comment:
The if statement is correct but the text in this warning is wrong. Just need to swap MT_ON and MT_OFF. 

That said I might rephrase it, what you've written is what a programmer would expect i.e. it explains the check we've done but an MR physicist/clinician probably wouldn't initially understand why we've added all the voxels up. Perhaps something like
`The average intensity of the MT on image is more than the average intensity of the MT off image, this will lead to negative MTR values which is not usually desirable. Please check you've input MTR on and MTR off in the correct order.`

Reply:
I can write that warning text and I understand the different perspectives. However, if we're mentioning "average intensity" in the warning text, would it be better to use `np.mean()` instead of `np.sum()`?

Reply:
The test is fundamentally the same, just scaled by a constant, but sure.

Reply:
Should be good to go now